src/common.sh

```diff
@@ -232,7 +232,8 @@ common::fixperms() {
 
     # Some distributions require CAP_DAC_OVERRIDE on several files and directories, fix these.
     # See https://bugzilla.redhat.com/show_bug.cgi?id=517575 for some context.
-    find "${path}" -maxdepth 5 \( -type d ! -perm -u+w -exec chmod -f u+w {} \+ \) -o \( ! -perm -u+r -exec chmod -f u+r {} \+ \)
+    chmod -f -R u+rX "${path}"
+    find "${path}" -maxdepth 5 -type d ! perm -u=w -exec chmod -f u+w {} \+
```

Comment:
The problem here is that `find` doesn't run the `exec` commands right away - it tries to traverse into the directory before editing it. It would be nice if we would simply be able to add a `u+r` to the original exec command for directories. And it's unfortunate that my solution edits the ctime for all files. But this is the best I could come up with.